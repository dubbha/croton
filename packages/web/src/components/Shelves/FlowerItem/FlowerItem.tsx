import React from 'react';
import { Card } from 'react-bootstrap';

import '../styles.scss';

type Props = {
  id: number;
  name: string;
  description: string;
  onSelect: (id: number) => void;
}

export const FlowerItem = ({
  id,
  name,
  description,
  onSelect,
}: Props) => (
  <div className="flower-item">
    <Card text="primary" onClick={() => onSelect(id)}>
      <Card.Img variant="top" src="" />
      <Card.Body>
        <Card.Title>{name}</Card.Title>
        <Card.Text>
          {description}
        </Card.Text>
      </Card.Body>
    </Card>
  </div>
);
