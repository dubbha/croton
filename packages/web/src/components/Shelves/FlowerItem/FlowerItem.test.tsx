import React from 'react';
import { render } from '@testing-library/react';
import { FlowerItem } from './FlowerItem';

describe('components/FlowerItem', () => {
  const props = {
    id: 0,
    name: '',
    location: '',
    description: '',
    addNew: false,
    onSelect: () => {},
  };

  it('should render successfully', () => {
    const { container } = render(<FlowerItem {...props} />);

    expect(container.firstChild).toMatchSnapshot();
  });
});
