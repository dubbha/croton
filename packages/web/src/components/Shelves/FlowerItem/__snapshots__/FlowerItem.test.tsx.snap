// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/FlowerItem should render successfully 1`] = `
<div
  class="flower-item"
>
  <div
    class="card text-primary"
  >
    <img
      class="card-img-top"
      src=""
    />
    <div
      class="card-body"
    >
      <div
        class="card-title h5"
      />
      <p
        class="card-text"
      />
    </div>
  </div>
</div>
`;
